Admin/AdminHelper.php

```diff
@@ -201,4 +229,105 @@ public function camelize($property)
     }
+
+    /**
+     * @param AdminInterface $admin
+     * @param array          $elements
+     *
+     * @return string
+     */
+    protected function getEntityClassName(AdminInterface $admin, $elements)
+    {
+        $element = array_shift($elements);
+        $associationAdmin = $admin->getFormFieldDescription($element)->getAssociationAdmin();
+        if (count($elements) == 0) {
+            return $associationAdmin->getClass();
+        } else {
+            return $this->entityClassNameFinder($associationAdmin, $elements);
+        }
+    }
+
+    /**
+     * get access path to element which works with PropertyAccessor.
+     *
+     * @param string $elementId
+     * @param mixed  $entity
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    public function getElementAccessPath($elementId, $entity)
+    {
+        $propertyAccessor = new PropertyAccessor();
+
+        $initial = preg_replace('#(_(\d+)_)#', '[$2]', implode('_', explode('_', substr($elementId, strpos($elementId, '_') + 1))));
+        $parts = preg_split('#\[\d+\]#', $initial);
+
+        $part_return_value = $return_value = '';
+        $current_entity = $entity;
+
+        foreach ($parts as $key => $value) {
+            $sub_parts = explode('_', $value);
+            $id = '';
+            $dot = '';
+
+            foreach ($sub_parts as $sub_value) {
+                $id .= ($id) ? '_'.$sub_value : $sub_value;
+
+                if ($this->pathExists($propertyAccessor, $current_entity, $part_return_value.$dot.$id)) {
+                    $part_return_value .= $dot.$id;
+                    $dot = '.';
+                    $id = '';
+                } else {
+                    $dot = '';
+                }
+            }
+
+            if ($dot !== '.') {
+                throw new \Exception(sprintf('Could not get element id from %s Failing part: %s', $elementId, $sub_value));
+            }
+
+            preg_match("#$value\[(\d+)#", $initial, $matches);
+
+            if (isset($matches[1])) {
+                $part_return_value .= '['.$matches[1].']';
+            }
+
+            $return_value .= $return_value ? '.'.$part_return_value : $part_return_value;
+            $part_return_value = '';
+
+            if (isset($parts[$key + 1])) {
+                $current_entity = $propertyAccessor->getValue($entity, $return_value);
+            }
+        }
+
+        return $return_value;
+    }
+
+    /**
+     * check if given path exists in $entity.
+     *
+     * @param PropertyAccessor $propertyAccessor
+     * @param mixed            $entity
+     * @param string           $path
+     *
+     * @return bool
+     *
+     * @throws \RuntimeException
+     */
+    private function pathExists(PropertyAccessor $propertyAccessor, $entity, $path)
+    {
+        if (method_exists($propertyAccessor, 'isReadable')) {
+            return $propertyAccessor->isReadable($entity, $path);
```

Comment:
In which case doesn't this exist? Please indicate it so that we can remove the piece of code below when dropping compatibility with old versions of symfony.